parity/snapshot.rs

```diff
@@ -170,7 +171,20 @@ impl SnapshotCommand {
 		execute_upgrades(&self.dirs.base, &db_dirs, algorithm, self.compaction.compaction_profile(db_dirs.db_root_path().as_path()))?;
 
 		// prepare client config
-		let client_config = to_client_config(&self.cache_config, Mode::Active, tracing, fat_db, self.compaction, self.wal, VMType::default(), "".into(), algorithm, self.pruning_history, true);
+		let client_config = to_client_config(
```

Comment:
the fact that this stuff is duplicated so much is practically a crime :)
Since basically every subcommand needs to initialize a client (and initializing a client needs, at a minimum, some of these parameters), the commands themselves should just take a client config rather than rebuilding it each and every time.